scripts/copy_demos.py

```diff
@@ -0,0 +1,61 @@
+import shutil
+import os
+import pathlib
+import textwrap
+import argparse
+
+
+def copy_all_demos(source_dir: str, dest_dir: str):
+    demos_to_copy = [
+        "blocks_essay",
+        "blocks_js_methods",
+        "blocks_layout",
+        "blocks_mask",
+        "blocks_multiple_event_triggers",
+        "calculator",
+        "fake_gan",
+        "gender_sentence_default_interpretation",
+        "image_mod_default_image",
+        "interface_parallel_load",
+        "interface_random_slider",
+        "interface_series_load",
+        "kitchen_sink",
+        "model3D",
+        "reverse_audio",
+        "sst_or_tts",
+        "stream_audio",
+        "stream_frames",
+        "zip_two_files",
+    ]
+    for demo in demos_to_copy:
+        shutil.copytree(
+            os.path.join(source_dir, demo),
+            os.path.join(dest_dir, demo),
+            dirs_exist_ok=True,
+        )
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="Copy all demos to all_demos and update requirements"
+    )
+    parser.add_argument("gradio_version", type=str, help="Gradio")
+    args = parser.parse_args()
+
+    source_dir = pathlib.Path(pathlib.Path(__file__).parent, "..", "demo")
+    dest_dir = pathlib.Path(
+        pathlib.Path(__file__).parent, "..", "demo", "all_demos", "demos"
+    )
+    copy_all_demos(source_dir, dest_dir)
+    reqs_file_path = pathlib.Path(
+        pathlib.Path(__file__).parent, "..", "demo", "all_demos", "requirements.txt"
+    )
+    requirements = f"""
+    {args.gradio_version}
```

Comment:
I recently experienced this,
In hugging face spaces defining gradio version in requirements.txt file does not make it work. You need to define it in README.md, see [here](https://huggingface.co/spaces/farukozderim/test-new-queue/blob/main/README.md#L7).

Also what do you think of using packages in test/requirements.txt instead of defining each manually?

Reply:
It'll work as long as the version in requirements.txt matches the latest stable version (what's listed in README by default). Currently working on a fix so that this whole process still works when you change `version.txt` in a release PR.

Reply:
@FarukOzderim Regarding packages in `test/requirements.txt` - If we were running all demos I think that may be a more appropriate solution but since we're only running a subset, I think manually specifying the minimal set of required packages is better (faster build with fewer packages).